scripts/kconfig/kconfigfunctions.py

```
@@ -932,6 +950,7 @@ def substring(kconf, _, string, start, stop=None):
         "dt_compat_enabled": (dt_compat_enabled, 1, 1),
         "dt_compat_on_bus": (dt_compat_on_bus, 2, 2),
         "dt_compat_any_has_prop": (dt_compat_any_has_prop, 2, 2),
+        "dt_compat_any_has_prop_val": (dt_compat_any_has_prop_val, 3, 3),
```

you can allow an optional `value` argument to the existing function by incrementing max number of args like this:
```suggestion
        "dt_compat_any_has_prop": (dt_compat_any_has_prop, 2, 3),
```